Library/Formula/llvm.rb

```diff
@@ -111,6 +120,37 @@ class Llvm < Formula
   fails_with :gcc
   fails_with :llvm
 
+  # Disable superenv if building with additional runtime libraries (compiler-rt)
+  #
+  # Something about superenv confuses the linker when compiling the support
+  # libraries (compiler-rt). Linking fails with errors similar to:
+  #   ...
+  #   ld: building for iOS simulator, but linking against dylib built for OSX,
+  #     file '/usr/lib/libc++.dylib' for architecture x86_64
+  #   clang: error: linker command failed with exit code 1 (use -v to see invocation)
+  #   make[2]: *** [lib/clang/3.8.0/lib/darwin/
+  #     libclang_rt.ubsan_iossim_dynamic.dylib] Error 1
+  #   ...
+  #   fatal error: /Applications/Xcode-beta.app/Contents/Developer/Toolchains/
+  #     XcodeDefault.xctoolchain/usr/bin/lipo: specifed architecture type (i386)
+  #     for file (/tmp/llvm20150910-69149-1cdqr2s/./lib/clang/3.8.0/lib/darwin/
+  #     libclang_rt.builtins_i386_iossim.a) does not match its cputype
+  #     (16777223) and cpusubtype (3) (should be cputype (7) and cpusubtype (3))
+  #   make[2]: *** [projects/compiler-rt/lib/builtins/
+  #     libclang_rt.builtins_iossim.a] Error 1
+  #   <cleanup and display error>
+  # Unfortunately, the reason the linker is doing this is not known at this
+  # time (11 September 2015). This is not the only incident of its kind; see:
+  #     https://forums.developer.apple.com/thread/13795
+  # so it is possible that this can be removed if/when XCode and/or the CLT get
+  # fixed.
+  #
+  # Builds without these support libraries completed normally, which is why
+  # this change was not required earlier. In addition, the libraries are only
+  # built when clang is built
+
+  env :std if build.with? "clang"
```

Comment:
Hmmm, ok. I'll take a closer look when I can. Here's what I (think I) know so far:

None of the makefiles I looked at had mixed platforms in their commands, so if it's a bad command it probably isn't local to that folder or surrounding folders. I didn't look at the ENV folder because I thought it just had symlinks. Didn't know it did its own processing. Don't know if that could have messed up; stdout looked ok, with no mixups, but I don't know if what is printed is before or after the command is started.

I'll do some messing around later. And just wondering, what gets passed to the ENV folder that requires so much processing? Symlinking that to the regular binary might help reveal the issue, but that's only if it's in a "normal" format.